src/components/TopNav.tsx

```diff
@@ -1,20 +1,322 @@
 import React from "react";
 import { Link } from "react-router-dom";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import MenuIcon from "@mui/icons-material/Menu";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import Divider from "@mui/material/Divider";
+import Drawer from "@mui/material/Drawer";
+import LightMode from "@mui/icons-material/LightModeOutlined";
+import DarkMode from "@mui/icons-material/DarkModeOutlined";
+import { useTheme } from "@mui/material/styles";
+import { styled } from "@mui/material/styles";
+import { themeBreak } from "./ThemeHandler";
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+export const ColorModeContext = React.createContext({ toggleColorMode: () => {""} });
+// Desktop Nav
+const Responsive = styled("div")(() => ({
+    padding: themeBreak.spacing(1),
+    display: "none",
+    // Match [sm, md)
+    //       [600px, 900px)
+    [themeBreak.breakpoints.between("sm", "md")]: {
+        display: "flex",
+    },
+    [themeBreak.breakpoints.up("md")]: {
+        display: "flex",
+    },
+}));
+// Mobile Nav 
+const SmallView = styled("div")(() => ({
+    padding: themeBreak.spacing(1),
+    [themeBreak.breakpoints.down("sm")]: {
+        display: "flex",
+    },
+    [themeBreak.breakpoints.up("sm")]: {
+        display: "none",
+    },
+}));
 
-const TopNav = () => {
+interface Props {
+    /**
+     * Injected by the documentation to work in an iframe.
+     * You won"t need it on your project.
+     */
+    window?: () => Window;
+}
+export default function TopNav (props: Props) {
+    // Get Current MUI Theme    
+    const theme = useTheme();   
+    const colorMode = React.useContext(ColorModeContext); 
+    const LinkColorSwitch = (
+        theme.palette.mode === "dark" ? (
+            "#F3FFDE"
+        ) : (
+            "#191919"
+        )
+    )
+    
+    const Links = {
+        color: LinkColorSwitch,
+        textDecoration: "none",
+        fontWeight: 700,
+    };
+
+    const { window } = props;
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const handleDrawerToggle = () => {
+        setMobileOpen(!mobileOpen);
+    };
+    // Theme Switch Icon Changes Based on Theme
+    const toggleButton = (
```

Comment:
This is good but can be included directly in the rendered code rather than separated like this. If you want to separate it then I suggest making is a separate component entirely. This way our component structure can be kept clear and consistent:
```
Imports

Validation schemas and special case code goes here...

const ComponentTitle (propsEtc) => {

    // Logic and state go here...

return (

    // Rendered content goes here...

)
}

Style object goes here...

```